pkg/cloudclient/cloudclient.go

```diff
@@ -5,42 +5,79 @@ import (
 	"fmt"
 	"time"
 
-	awscredsv2 "github.com/aws/aws-sdk-go-v2/credentials"
-	awscredsv1 "github.com/aws/aws-sdk-go/aws/credentials"
 	ocmlog "github.com/openshift-online/ocm-sdk-go/logging"
-	awsCloudClient "github.com/openshift/osd-network-verifier/pkg/cloudclient/aws"
-	gcpCloudClient "github.com/openshift/osd-network-verifier/pkg/cloudclient/gcp"
 	"github.com/openshift/osd-network-verifier/pkg/output"
+	"github.com/openshift/osd-network-verifier/pkg/parameters"
+	"github.com/openshift/osd-network-verifier/pkg/utils"
+)
+
+// common commandline args
+type CmdOptions struct {
```

Comment:
The cloudclient isn't directly tied to the CLI inputs and should not contain the options struct for CLI interfaces.

Reply:
I see exactly what you're saying. I redid this.
Removed these inputs. 
Defined client configs in infrastructure and execution configs in cloudConfig.
All other places use pointers to these structs and they're not combined anywhere or duplicated.
The structs now look deeper and more nested, but we have better modularity and separation of concerns.